src/app/clusters/access-control-server/access-control-server.cpp

```diff
@@ -15,4 +15,423 @@
  *    limitations under the License.
  */
 
-void MatterAccessControlPluginServerInitCallback() {}
+#include <access/AccessControl.h>
+#include <access/examples/ExampleAccessControlDelegate.h>
+
+#include <app-common/zap-generated/af-structs.h>
+#include <app-common/zap-generated/cluster-objects.h>
+#include <app-common/zap-generated/enums.h>
+
+#include <app/AttributeAccessInterface.h>
+#include <app/CommandHandler.h>
+#include <app/ConcreteCommandPath.h>
+#include <app/data-model/Encode.h>
+#include <app/util/af.h>
+#include <app/util/attribute-storage.h>
+
+using namespace chip;
+using namespace chip::app;
+using namespace chip::Access;
+
+namespace AccessControlCluster = chip::app::Clusters::AccessControl;
+
+struct AccessControlEntryCodec
+{
+    static CHIP_ERROR Convert(AuthMode from, AccessControlCluster::AuthMode & to)
```

Comment:
This to me suggests  that Access::AuthMode should be a type alias for AccessControlCluster::AuthMode, so all we need to do here is check that we're in range....

Reply:
That would make the code a tiny bit smaller (how much? a few small switch statements would go away). But eventually (when we get beyond features and into optimization) I want to bit encode the auth mode and privilege together. That will optimize for the common runtime check case, whereas managing ACL entries occurs much less often.